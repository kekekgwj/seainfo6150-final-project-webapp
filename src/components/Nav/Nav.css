.global-nav {
    position: relative;
    height: 28px;
    color: #d5d5d5;
    background-color: #545652;
    min-width: 850px;
    font-size: 24px;
    zoom: 1;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 20px 0 20px;
}

.global-nav-items ul{
    margin: 0;
    padding: 0;
    list-style: none;
}
.global-nav-items li {
    display: inline-block;
    zoom: 1;
    line-height: 28px;
    font-size: 24px;
    vertical-align: baseline;
    color: #ddd;
}
.global-nav-items a {
    color: #ddd;
    text-decoration: none;
}
.top-nav-user {
    font-size: 16px;
}

.top-nav-user div {
    display: inline-block;
}
.top-nav-user span{
    cursor: pointer;
}
.top-nav-user img {
    height: 16px;
    width: 16px;
    padding-right: 5px;
    position: relative;
    top:3px;
}
.nav-log-out {
    padding-left: 10px;
    position: relative;
    left: 10px;
    border-left: 1px #ddd solid;
}
.nav-log-out  img {
    position: relative;
    left: 10px;
}