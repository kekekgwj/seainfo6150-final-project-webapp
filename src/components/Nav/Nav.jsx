import React from "react";
import { Link } from "react-router-dom";
import './Nav.css';
import logOut from "../../assets/log-out.png"
import logIn from "../../assets/login-icon.png";
import userIcon from "../../assets/login-user.png";
import { useHistory } from 'react-router-dom';

const Nav = ({username, onLogOut}) => {
    const loggedEle = (                
        <div>
            <div onClick={() => toHomePage()}>
                <span className="nav-user-icon">
                    <img src={userIcon} alt="user-icon"/>
                </span>
                <span>
                    {username}
                </span> 
            </div>
            <div className="nav-log-out" onClick={onLogOut}>
                <span>Log out</span>
                <span className="nav-user-icon">
                    <img src={logOut} alt="log out"/>
                </span>
            </div> 
        </div>);
    const toLoginEle = (
        <div onClick={() => toLoginPage()}>
            <span className="nav-user-icon"> 
                <img src={logIn} alt="login"/>
            </span>
            <span>Log In</span>
        </div>
    ); 

    const history = useHistory();
    const toHomePage = () => {
        history.push('/home');
    }
    const toLoginPage = () => {
        history.push('/login')
    }
    return (
        <div className="global-nav">
            <div className="global-nav-items">
                <ul>
                    <li>
                        <Link to="/movie_list">Movies</Link>
                    </li>
                </ul>
            </div>
            <div className="top-nav-user">
                { username ? loggedEle : toLoginEle }
            </div>
        </div>
     )
}

export default Nav;