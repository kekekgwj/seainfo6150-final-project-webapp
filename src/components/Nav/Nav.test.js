import React from "react";
import Nav from "./Nav";


describe("nav tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Nav username='test user'/>);
    expect(container).toMatchSnapshot();
  });
});