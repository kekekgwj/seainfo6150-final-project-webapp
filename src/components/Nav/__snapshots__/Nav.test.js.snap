// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav tests renders correctly 1`] = `
<div>
  <div
    class="global-nav"
  >
    <div
      class="global-nav-items"
    >
      <ul>
        <li>
          <a
            href="/movie_list"
          >
            Movies
          </a>
        </li>
      </ul>
    </div>
    <div
      class="top-nav-user"
    >
      <div>
        <div>
          <span
            class="nav-user-icon"
          >
            <img
              alt="user-icon"
              src="login-user.png"
            />
          </span>
          <span>
            test user
          </span>
        </div>
        <div
          class="nav-log-out"
        >
          <span>
            Log out
          </span>
          <span
            class="nav-user-icon"
          >
            <img
              alt="log out"
              src="log-out.png"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
